src/collision_func_matrix.cpp

```diff
@@ -82,34 +82,20 @@ std::size_t ShapeShapeCollide(const CollisionGeometry* o1, const Transform3f& tf
   FCL_REAL distance = ShapeShapeDistance <T_SH1, T_SH2>
     (o1, tf1, o2, tf2, nsolver, distanceRequest, distanceResult);
 
-  if (distance <= 0) {
-    if (result.numContacts () < request.num_max_contacts) {
-      const Vec3f& p1 = distanceResult.nearest_points [0];
-      const Vec3f& p2 = distanceResult.nearest_points [1];
+  const Vec3f& p1 = distanceResult.nearest_points [0];
+  const Vec3f& p2 = distanceResult.nearest_points [1];
+  FCL_REAL distToCollision = distance - request.security_margin;
 
-      Contact contact (o1, o2, distanceResult.b1, distanceResult.b2,
-          (p1+p2)/2,
-          distanceResult.normal,
-          -distance+request.security_margin);
+  internal::updateDistanceLowerBoundFromLeaf (request, result, distToCollision, p1, p2);
+  if (distToCollision <= 0 && result.numContacts () < request.num_max_contacts) {
```

Comment:
Currently, this results is not taking into account the break_distance information.
Compared to other collision checks.
```suggestion
  if (distToCollision <= request.break_distance && result.numContacts () < request.num_max_contacts) {
```

Reply:
No, there is a contact only if distToCollision <= 0. At the moment, `break_distance` is not useful for shape-shape pairs.